src/layout/layout.rs

```diff
@@ -426,34 +437,83 @@ impl Layout {
 
         // create an element for each constraint that needs to be applied. Each element defines the
         // variables that will be used to compute the layout.
-        let elements = layout
+        let elements: Vec<Element> = layout
             .constraints
             .iter()
-            .map(|_| Element::new())
-            .collect::<Vec<Element>>();
-
-        // ensure that all the elements are inside the area
-        for element in &elements {
-            solver.add_constraints(&[
-                element.start | GE(REQUIRED) | area_start,
-                element.end | LE(REQUIRED) | area_end,
-                element.start | LE(REQUIRED) | element.end,
-            ])?;
-        }
+            .map(|_| Element::constrain(&mut solver, (area_start, area_end)))
+            .try_collect()?;
+
         // ensure there are no gaps between the elements
         for pair in elements.windows(2) {
             solver.add_constraint(pair[0].end | EQ(REQUIRED) | pair[1].start)?;
         }
-        // ensure the first element touches the left/top edge of the area
-        if let Some(first) = elements.first() {
-            solver.add_constraint(first.start | EQ(REQUIRED) | area_start)?;
-        }
-        if layout.segment_size != SegmentSize::None {
-            // ensure the last element touches the right/bottom edge of the area
-            if let Some(last) = elements.last() {
-                solver.add_constraint(last.end | EQ(REQUIRED) | area_end)?;
+
+        match layout.flex {
+            Flex::StretchLast => {
+                if let Some(first) = elements.first() {
+                    solver.add_constraint(first.start | EQ(REQUIRED) | area_start)?;
+                }
+                if let Some(last) = elements.last() {
+                    solver.add_constraint(last.end | EQ(REQUIRED) | area_end)?;
+                }
+            }
+            Flex::Stretch => {
+                if let Some(first) = elements.first() {
+                    solver.add_constraint(first.start | EQ(REQUIRED) | area_start)?;
+                }
+                if let Some(last) = elements.last() {
+                    solver.add_constraint(last.end | EQ(REQUIRED) | area_end)?;
+                }
+                // prefer equal chunks if other constraints are all satisfied
+                for (left, right) in elements.iter().tuple_combinations() {
+                    solver.add_constraint(left.size() | EQ(WEAK) | right.size())?;
+                }
+            }
+            Flex::Center => {
+                let flex_start_element = Element::constrain(&mut solver, (area_start, area_end))?;
+                let flex_end_element = Element::constrain(&mut solver, (area_start, area_end))?;
+                solver.add_constraint(
+                    flex_start_element.size() | EQ(STRONG) | flex_end_element.size(),
```

Comment:
Yes it does change, because we want this to be exactly equal to `Min` and `Max` when they are at their limit for good solutions. 

In this PR I didn't change any weights of any constraints but in another PR I will change weights which affect how constraints behave.